### PR TITLE
Add support for Python 3.11.0a7.

### LINKED ISSUE #67: Use PyFloat_Pack8() on Python 3.11 alpha7
The private _PyFloat_Pack8() and _PyFloat_Unpack8() functions were removed in Python 3.11, but Python 3.11 alpha7 adds new clean public PyFloat_Pack8() and PyFloat_Unpack8() functions: see https://bugs.python.org/issue46906

* https://github.com/zopefoundation/zodbpickle/blob/master/src/zodbpickle/_pickle_33.c#L1695
* https://github.com/zopefoundation/zodbpickle/blob/master/src/zodbpickle/_pickle_33.c#L4247

zodbpickle should use them to be compatible with Python 3.11.

### PR DESCRIPTION
**Pull request by icemac**
To run successfully the following changes are required:

- [x] ~~merge of https://github.com/zopefoundation/zodbpickle/pull/64~~
- [x] recreate the changes made in https://github.com/zopefoundation/zodbpickle/pull/64
- [x] fix of https://github.com/zopefoundation/zodbpickle/pull/64#issuecomment-1025719735 see #67 for the solution 
- [x] GHA must provide Python 3.11.0a7.

Fixes #67.

### CODE REVIEW

**Comment by icemac** [comment]
Locally all tests ran fine. Now we have to wait for 3.11.0a7 on GHA.

**Comment by dataflake** [comment]
@icemac Looks like the only tests that show as failing are old runs from a week ago, everything else succeeds.

**Comment by icemac** [comment]
This PR is now ready for review.

File: CHANGES.rst [hunk 1]
```diff
@@ -5,7 +5,7 @@
 2.2.1 (unreleased)
 ==================
 
-- Nothing changed yet.
+- Add support for Python 3.11 (as of 3.11.0a4).
```
Comment (dataflake) [inline CHANGES.rst:8]:
```suggestion
- Add support for Python 3.11 (as of 3.11.0a7).
```

**Review by dataflake — APPROVED** [review]
Just one small change

**Comment by icemac** [comment]
Thank you for reviewing this PR. 😃

**Pull request merged**